great_expectations/core/batch_config.py

```diff
@@ -18,24 +19,15 @@ class BatchConfig(pydantic.BaseModel):
     TODO: Add splitters and sorters?
     """
 
-    class Config:
-        arbitrary_types_allowed = True
-
-    context: AbstractDataContext
+    id: Optional[str] = None
     name: str
-    datasource_name: str
-    data_asset_name: str
-
-    @property
-    def data_asset(self) -> DataAsset:
-        """Get the DataAsset referenced by this BatchConfig."""
-        datasource = self.context.get_datasource(self.datasource_name)
-        assert isinstance(datasource, Datasource)
-
-        return datasource.get_asset(self.data_asset_name)
+    data_asset: DataAsset
 
     def build_batch_request(
         self, batch_request_options: Optional[BatchRequestOptions] = None
     ) -> BatchRequest:
         """Build a BatchRequest from the asset and batch request options."""
         return self.data_asset.build_batch_request(options=batch_request_options)
+
+    def save(self) -> None:
+        self.data_asset._save_batch_config(self)
```

Comment:
we're an external caller to the `DataAsset`, i would recommend `DataAsset.save_batch_config` be public (although not `@public_api`).